### PR TITLE
feat: add `pixi clean` command

### PR DESCRIPTION
**Pull request by ruben-arts**
Initial proposal
Includes
- `pixi clean` cleans environments
- `pixi clean -e prod` cleans a environment
- `pixi clean cache` cleans cache dir
- `pixi clean cache --conda/--pypi` cleans a specific package cache

fixes: #1321

### CODE REVIEW

**Comment by ruben-arts** [comment]
I'm not happy with the cli yet. I want `env` to be the default and `cache` being a sub command.

File: src/cli/clean.rs [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+#[derive(Parser, Debug)]
+pub struct CacheArgs {
+    /// Clean only the pypi related cache.
+    #[arg(long)]
+    pub pypi: bool,
+
+    /// Clean only the conda related cache.
+    #[arg(long)]
+    pub conda: bool,
+    // TODO: Would be amazing to have a --unused flag to clean only the unused cache.
```
Comment (nichmor) [inline src/cli/clean.rs:43]: just my 2 cents - we need to be sure that we can support windows with this. I think they don't have inode count

File: src/cli/clean.rs [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    pb.enable_steady_tick(Duration::from_millis(100));
+    pb.set_style(long_running_progress_style());
+    pb.set_message(format!(
+        "{} {}",
+        console::style("Removing").green(),
+        folder.clone().display()
+    ));
+
+    // Ignore errors
+    let _ = tokio::fs::remove_dir_all(&folder).await;
```
Comment (nichmor) [inline src/cli/clean.rs:136]: do we want to log the errors ? ( maybe with debug?)

File: src/cli/clean.rs [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+                                .to_str()
+                                .expect("expected to have a manifest_path")
+                        )
+                    })
+                })
+                .transpose()?;
+
+            if let Some(explicit_env) = explicit_environment {
+                remove_folder_with_progress(explicit_env.dir(), true).await?;
+                tracing::info!("Skipping removal of task cache and solve group environments for explicit environment '{:?}'", explicit_env.name());
```
Comment (nichmor) [inline src/cli/clean.rs:74]: let's add the same tracing:info for `else` branch

**Comment by bollwyvl** [comment]
`pixi clean task <taskname>` would also be very helpful. 

In some systems (e.g. `doit`), `clean` can be optionally configured to remove:
- remove all `outputs`
  - `clean = true`
- some specific set of files
  - `clean = ["*.tsbuildinfo", "lib"]`
- the `clean` of all `depends-on`
  - `clean-depends-on = true`
    - though maybe this would be better when a `depends-on` can be a more full task with
      - `depends-on = [{task="build", environment="build", clean=true}]`

File: src/cli/clean.rs (hunk 3 above)
Comment (ruben-arts) [reply src/cli/clean.rs:74]: It's not really skipping anything in that part so there everything would be logged by either the progress bar or other info's

**Comment by ruben-arts** [comment]
@bollwyvl I like this idea!

**Comment by ruben-arts** [comment]
@bollwyvl I gave task a try but got burned out by the complexity of the glob work. Still think it's a good idea but I want to get the simple version in first.

**Review by nichmor — APPROVED** [review]
looks good!

**Review by baszalmstra — APPROVED** [review]

**Pull request merged**